docker/docker_client.go

```diff
@@ -55,6 +58,7 @@ type bearerToken struct {
 // dockerClient is configuration for dealing with a single Docker registry.
 type dockerClient struct {
 	ctx             *types.SystemContext
+	httpWrapper     httpWrapper // nil unless running tests
```

Comment:
> I still believe, in Golang, it's better to have an interface for this where it could be mocked in testing. 

An interface for which “this”? `dockerClient` or `http.*`?

(The `httpWrapper` does allow substituting a completely fake `http.RoundTripper`, it does not require the use of `go-vcr`.)

Mocking `dockerClient` is a bit awkward in that the authentication behavior is one of the essential components to be tested. Though i can see the argument that there should be a separate test set for the authentication, which uses a real `dockerClient`, and a separate test set which assumes sufficient authentication and just tests the individual read/write operations, which uses a mock `dockerClient` for simplicity. (I am not _sure_ that it can be safely factored that way, but I can see how it is attractive.)

[Also, #255 adds helper methods shared between the source/destination to `dockerClient`, which would make the mocking of `dockerClient` awkward; but they could just as well be methods which take a `dockerClient` parameter.]

> This could be far better instead but far from what I think golang tests for this should be.

I guess let's have a conversation about what the tests should be? on IRC, or perhaps over the phone?

I am honestly unsure about this approach. I do like that it is pretty close to real-world request/responses, and that it is easy to write the tests (just write client calls, the real server will provide real responses); OTOH there is substantial risk that maintaining the recordings will be a pain.